### PR TITLE
Add reannotate function to combiner

### PR DESCRIPTION
**Pull request by chrisvittal**
This joins the output of the combiner with the output some of of GATK's
post joint-calling tools. It is intended that this function be run with some
combiner output, the rows of the combiner output run through `summarize`, 
and GATKs output from a sites only VCF produced from that `summarize` output.

It also adds a little documentation.

### CODE REVIEW

File: hail/python/hail/experimental/vcf_combiner.py [hunk 1]
```diff
@@ -165,8 +170,69 @@ def summarize(mt):
...
+            QUALapprox: int32,
+            RAW_MQ: float64,
+            ReadPosRankSum: float64,
+            SB_TABLE: array<int32>,
+            SOR: float64,
+            VarDP: int32
+        }
+    ```
+
+    NOTE: You will not be able to run `combine_gvcfs` with the output of this
```
Comment (jigold) [inline hail/python/hail/experimental/vcf_combiner.py]: Can you change `Note:` to this:

```
Note
-----
You will....
```

**Comment by chrisvittal** [comment]
You can take another look, but don't approve yet, I've been told there's more coming. The initial that I received and based this PR off of is not complete.

**Comment by chrisvittal** [comment]
@jigold this should be good now

File: hail/python/hail/experimental/vcf_combiner.py [hunk 2]
```diff
@@ -142,6 +142,13 @@ def lgt_to_gt(lgt, la):
 
 
 def summarize(mt):
+    """Computes summary statistics
+
+    Note
+    ----
+    You will not be able to run `combine_gvcfs` with the output of this
```
Comment (jigold) [inline hail/python/hail/experimental/vcf_combiner.py]: Sorry. Just saw that `combine_gvcfs` is not linked to the function. i.e. `:func:...`. See other instances below.

**Comment by chrisvittal** [comment]
I added the links.

**Pull request merged**